compiler/rustc_hir_typeck/src/check.rs

```diff
@@ -44,10 +44,11 @@ pub(super) fn check_fn<'a, 'tcx>(
 
     let declared_ret_ty = fn_sig.output();
 
+    let body_def_id = tcx.hir().local_def_id(body.value.hir_id);
```

Comment:
This will ICE: bodies' HirId do not have a `LocalDefId`.  Their `body_owner` does, accessible using `tcx.hir().body_owner_def_id(body_id)`.